### PR TITLE
Adding programmatic_descriptions to table search

### PR DESCRIPTION
**Pull request by samshuster**
### Summary of Changes

This adds programmatic_description nodes as part of search.

### Tests

Modified existing tests.

### Documentation

None

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

**Comment by samshuster** [comment]
hey @jinhyukchang now that all the frontend changes have been merged, would you be ready to take a look at this?

**Comment by samshuster** [comment]
hey @feng-tao I fixed the merge conflicts on this one and it should be good to go again.

File: tests/unit/proxy/test_elasticsearch.py [hunk 1]
```diff
@@ -93,7 +95,8 @@ def setUp(self) -> None:
                                              column_names=['test_col1', 'test_col2'],
                                              tags=self.mock_empty_tag,
                                              badges=self.mock_empty_badge,
-                                             last_updated_timestamp=1527283287)
+                                             last_updated_timestamp=1527283287,
+                                             programmatic_descriptions=[])
```
Comment (jinhyukchang) [inline tests/unit/proxy/test_elasticsearch.py:99]: Could you add a test case where programmatic_descriptions is `None` or not being assigned to see if it is backward compatible?

Comment (samshuster) [reply tests/unit/proxy/test_elasticsearch.py:99]: ok will do, good call out.

Comment (samshuster) [reply tests/unit/proxy/test_elasticsearch.py:99]: Ok added a case where it is None as well just in case. Also confirmed in the table.py programmatic_descriptions defaults to [] not None.

Comment (samshuster) [reply tests/unit/proxy/test_elasticsearch.py:99]: hmm ok, no that won't work because the type of programmatic_description is list[string]. It has a default value of [] if not set. Also, this test uses mock search results.

File: tests/unit/proxy/test_elasticsearch.py [hunk 2]
```diff
@@ -530,6 +533,7 @@ def test_create_document(self, mock_uuid: MagicMock) -> None:
                 'tags': [],
                 'badges': [],
                 'total_usage': 0,
+                'programmatic_descriptions': [],
```
Comment (jinhyukchang) [inline tests/unit/proxy/test_elasticsearch.py:536]: I wonder what happens if `programmatic_descriptions` is not provided by Elasticsearch.

File: tests/unit/proxy/test_elasticsearch.py [hunk 3]
```diff
@@ -93,7 +95,8 @@ def setUp(self) -> None:
                                              column_names=['test_col1', 'test_col2'],
                                              tags=self.mock_empty_tag,
                                              badges=self.mock_empty_badge,
-                                             last_updated_timestamp=1527283287)
+                                             last_updated_timestamp=1527283287,
+                                             programmatic_descriptions=[])
 
         self.mock_result2 = MockSearchResult(name='test_table2',
```
Comment (samshuster) [inline tests/unit/proxy/test_elasticsearch.py:101]: hey @jinhyukchang here is a mock result where programmatic_descriptions aren't specified

File: tests/unit/proxy/test_elasticsearch.py (hunk 2 above)
Comment (samshuster) [reply tests/unit/proxy/test_elasticsearch.py:536]: ok sorry, thought i was on the right track. So currently, the models.table.py has this:
```programmatic_descriptions: List[str] = []```
which means that by default it will be an empty list.

Above, the first Table actually does not specify any programmatic_descriptions. Would this be the way to test no programmatic_descriptions being returned by elastic-search? Would a brand new unit-test be useful here? Let me know what you think.
